projects/igniteui-angular/grids/core/src/services/pdf/pdf-exporter-grid.spec.ts

```diff
@@ -0,0 +1,515 @@
+            childGrid.expandChildren = true;
+        });
+        fix.detectChanges();
+
+        exporter.exportEnded.pipe(first()).subscribe(() => {
+            expect(ExportUtilities.saveBlobToFile).toHaveBeenCalledTimes(1);
+            done();
+        });
+
+        exporter.export(grid, options);
+    });
+
+    it('should export tree grid with hierarchical data', (done) => {
+        TestBed.configureTestingModule({
+            imports: [
+                NoopAnimationsModule,
+                IgxTreeGridSortingComponent
+            ]
+        }).compileComponents();
+
+        const fix = TestBed.createComponent(IgxTreeGridSortingComponent);
+        fix.detectChanges();
+
+        const grid = fix.componentInstance.treeGrid;
+
+        exporter.exportEnded.pipe(first()).subscribe(() => {
+            expect(ExportUtilities.saveBlobToFile).toHaveBeenCalledTimes(1);
+            done();
+        });
+
+        exporter.export(grid, options);
+    });
+
+    it('should export tree grid with flat self-referencing data', (done) => {
+        TestBed.configureTestingModule({
+            imports: [
+                NoopAnimationsModule,
+                IgxTreeGridPrimaryForeignKeyComponent
+            ]
+        }).compileComponents();
+
+        const fix = TestBed.createComponent(IgxTreeGridPrimaryForeignKeyComponent);
+        fix.detectChanges();
+
+        const grid = fix.componentInstance.treeGrid;
+
+        exporter.exportEnded.pipe(first()).subscribe(() => {
+            expect(ExportUtilities.saveBlobToFile).toHaveBeenCalledTimes(1);
+            done();
+        });
+
+        exporter.export(grid, options);
+    });
+
+    it('should truncate long header text with ellipsis in multi-column headers', (done) => {
+        TestBed.configureTestingModule({
+            imports: [
+                NoopAnimationsModule,
+                BlueWhaleGridComponent
+            ]
+        }).compileComponents();
+
+        const fix = TestBed.createComponent(BlueWhaleGridComponent);
+        fix.detectChanges();
+
+        const grid = fix.componentInstance.grid;
+
+        exporter.exportEnded.pipe(first()).subscribe((args) => {
+            expect(ExportUtilities.saveBlobToFile).toHaveBeenCalledTimes(1);
+            // The PDF should be created successfully even with long header text
+            expect(args.pdf).toBeDefined();
+            done();
+        });
+
+        // Use smaller page size to force truncation
+        options.pageSize = 'a5';
+        exporter.export(grid, options);
+    });
+
+    describe('Pivot Grid PDF Export', () => {
+        let pivotGrid: IgxPivotGridComponent;
+        let fix;
+        beforeEach(async () => {
+            fix = TestBed.createComponent(IgxPivotGridMultipleRowComponent);
+            fix.detectChanges();
+            await wait();
+
+            pivotGrid = fix.componentInstance.pivotGrid;
+            spyOn(ExportUtilities as any, 'saveBlobToFile');
```

Comment:
This is duplicate of the main `describe('PDF Grid Exporter'`'s `beforeEach` that already spies this method and that applies to all tests in this suite and sub-suites, so the error is right and it should be removed:
```suggestion
```